### PR TITLE
OKTETO_HOME to override .okteto folder not $HOME

### LINKED ISSUE #1104: OKTETO_HOME should point to the okteto folder and not the HOME directory
**Is your feature request related to a problem? Please describe.**

I would like to relocate my OKTETO folder outside my home directory

Currently if I change the OKTETO_HOME environment variable, this relocates the `.okteto` folder but also makes all the logic to find `.ssh/config` and `.kube/config` inside the OKTETO_HOME folder.

**Describe the solution you'd like**

Currently `OKTETO_HOME` is used to determine the HOME directory:
https://github.com/okteto/okteto/blob/master/pkg/config/config.go#L89

I think OKTETO_HOME should instead be used in the `GetOktetoHome` function
https://github.com/okteto/okteto/blob/master/pkg/config/config.go#L49

**Describe alternatives you've considered**
None

**Additional context**

### PR DESCRIPTION
**Pull request by rberrelleza**
Fixes #1104

## Proposed changes
- OKTETO_HOME now points to the replacement for the .okteto folder
- Added an OKTETO_USERHOME for those who want to replace the $HOME directory (we've run into this in the past with minimalistic setups)

This is a breaking change, so we need to document it in the release notes.

### CODE REVIEW

**Comment by pchico83** [comment]
@rberrelleza is the vscode plugin using `$OKTETO_HOME`?

File: pkg/config/config.go [hunk 1]
```diff
@@ -86,7 +95,11 @@ func GetSyncthingLogFile(namespace, name string) string {
 
 // GetUserHomeDir returns the OS home dir
 func GetUserHomeDir() string {
-	if v, ok := os.LookupEnv("OKTETO_HOME"); ok {
+	if v, ok := os.LookupEnv("OKTETO_USERHOME"); ok {
```
Comment (pchico83) [inline pkg/config/config.go:98]: what is the use case for `OKTETO_USERHOME`?

Comment (rberrelleza) [reply pkg/config/config.go:98]: when we can't determine the path to the $HOME directory. that was the reason why we put OKTETO_HOME in the first place.

Comment (pchico83) [reply pkg/config/config.go:98]: but why would a user set `OKTETO_USERHOME` instead of `OKTETO_HOME`?

Comment (rberrelleza) [reply pkg/config/config.go:98]: we need to know where's the $HOME directory to be able to write the ssh configuration and kubeconfig files

**Review by pchico83 — APPROVED** [review]

**Comment by rberrelleza** [comment]
> @rberrelleza is the vscode plugin using `$OKTETO_HOME`?

no, I don't think so

File: pkg/config/config.go [hunk 2]
```diff
@@ -47,6 +48,14 @@ func GetBinaryFullPath() string {
 
 // GetOktetoHome returns the path of the okteto folder
 func GetOktetoHome() string {
+	if v, ok := os.LookupEnv("OKTETO_HOME"); ok {
```
Comment (pchico83) [inline pkg/config/config.go:51]: Could we call this one `OKTETO_FOLDER` and keep the current behavior for `OKTETO_HOME` to avoid breaking changes?

Comment (rberrelleza) [reply pkg/config/config.go:51]: sure

**Pull request merged**